game-app/game-core/src/main/java/games/strategy/engine/data/Rule.java

```diff
@@ -0,0 +1,35 @@
+package games.strategy.engine.data;
+
+import org.triplea.java.collections.CollectionUtils;
+import org.triplea.java.collections.IntegerMap;
+
+/**
+ * Superclass for {@link games.strategy.engine.data.RepairRule} and {@link
+ * games.strategy.engine.data.ProductionRule}.
+ */
+public interface Rule {
```

Comment:
You can use the `@ChangeOnNextMajorRelease` annotation here and mention that this can become an abstract base class in the future.

Reply:
Good to know. [I added the annotation](https://github.com/triplea-game/triplea/pull/10378/files#diff-1fc5c20175ae8cf69f7c6fbd1fe7a7a4d97b99fdf082de528070e5eed086635dR15).